src/quest_manager/models.py

```diff
@@ -19,12 +19,52 @@
 from tags.models import TagsModelMixin
 
 
+class CategoryQuerySet(models.QuerySet):
+    """
+    Custom queryset for Category model (campaigns).
+    """
+    def with_available_quests(self):
+        """
+        Annotates and filters categories that are active and have at least one
+        quest that is visible to students and not archived.
+        Uses a correlated subquery with EXISTS for performance.
+        """
+        filtered_quests = Quest.objects.filter(
```

Comment:
There should be an existing Quest manager method for this that is already tested, if not, create it there and call it. That  helps us keep Quest logic in the Quest model and can reuse it later, and also properly unit test it.

Also, "filtered_quests" is a terrible name because it tells you nothing about what the queryset is for or contains!

Reply:
Also, it's not clear what you mean by "available" since that's not a field.  There are a thousand ways quests could be available or unavailable, so be explicit here.  By default, when I read "available" it means available to a student, which doesn't make sense here.

What's this method for?  Maybe it needs the context added to the docstring.